src/components/ui/Utilities/memoize.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export const memoize = (fn: Function) => {
+  const map = {};
+  return (...args: any) => {
+    let n = args[0];
+    if (n in map) {
```

Comment:
Seems a little strange to accept multiple args but then memoize based on only the first arg. This likely will lead to bugs in any function memoizes but takes in multiple args

We either need to restrict to a single argument, or add support for mluti args (such as multi-arg index, or build the cache around the # of arguments passed)